src/components/organisms/Page.js

```diff
@@ -33,10 +50,57 @@ export function Page(props) {
   };
 
   useEffect(() => {
-    if (keycloak.authenticated && Object.keys(userProfileData).length === 0) {
+    if (
+      keycloak.authenticated &&
+      !triedFetchUserData &&
+      !isFetchingUserData &&
+      !fetchUserDataFailed
+    ) {
       dispatch(getUserData(keycloak));
+      setTriedFetchUserData(true);
+    }
+  }, [
+    keycloak,
+    dispatch,
+    fetchUserDataFailed,
+    isFetchingUserData,
+    triedFetchUserData,
+    setTriedFetchUserData,
+  ]);
+
+  useEffect(() => {
+    if (
+      triedFetchUserData &&
+      !fetchUserDataFailed &&
+      triedAnswersFromLocalStorageFailed
+    ) {
+      dispatch(
+        setAnswerActionCreator(
+          "province",
+          userProfileData.personAddressProvince
+        )
+      );
+      userProfileData.personGender === "SX1"
+        ? dispatch(setAnswerActionCreator("gender", "male"))
+        : dispatch(setAnswerActionCreator("gender", "female"));
+    }
```

Comment:
I thought we were going to overwrite user data with answers, not the other way around.

Reply:
It tries to get the answers if they exist in storage, and only if it can't does it use the answers from the user profile. If there are answers saved it shouldn't overwrite them.

Reply:
But it always needs the user profile + answers.

Reply:
ohh, I missed the triedAnswersFromLocalStorageFailed